### ISSUE TITLE
search-engine: "command-execute: Symbol's function definition is void: engine/search-google"

### ISSUE THREAD

**Issue by nicbet** [syl20bnr/spacemacs#6807]
#### Description
In `dotspacemacs/user-config()`, I added the following key-binding:
```
;; Custom keybinding to access google search
  (spacemacs/set-leader-keys "so"  'engine/search-google)
```
When a fresh instance of Emacs is started, the keybinding will throw error:
```
command-execute: Symbol's function definition is void: engine/search-google
```
However, after pressing `SPC a /` once and selecting any search engine, the key binding will work - it appears that the `engine/search-` functions are generated by invoking `SPC a /` and do not exist before. It would be helpful to use a key binding as above without having to initialize the `engine/search-` functions first through `SPC a /`.

#### Reproduction guide
- Start Emacs
- Press `SPC s o` 

*Observed behaviour:*
Error:
```
command-execute: Symbol's function definition is void: engine/search-google
```

*Expected behaviour:*
Entry field opens to input search terms

#### System Info
- OS: darwin
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: master (rev. 0283f64)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(erlang osx html colors themes-megapack
        (auto-completion :variables auto-completion-return-key-behavior nil auto-completion-tab-key-behavior 'complete :disabled-for org)
        better-defaults emacs-lisp git ess elixir ruby python unimpaired yaml javascript react markdown org
        (shell :variables shell-default-height 30 shell-default-term-shell "/bin/zsh" shell-default-position 'bottom)
        spell-checking syntax-checking version-control search-engine)
```

**Comment by TheBB** [comment]
Unfortunately it's difficult to do this without forcing `engine-mode` to load at startup. A better solution is to define the functions you need as autoloadable.

    (autoload 'engine/search-google "engine-mode")

We can probably do this in Spacemacs.

**Comment by TheBB** [comment]
Fixed by #6808

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
